src/dist/replication/common/fs_manager.h

```diff
@@ -50,16 +50,31 @@ struct dir_node
     std::string full_dir;
     int64_t disk_capacity_mb;
     int64_t disk_available_mb;
-    int64_t disk_available_ratio;
+    int disk_available_ratio;
+    // disk_density = disk_available_ratio - total_available_ratio, it's used to evaluate whether a
```

Comment:
Copy from hdfs.

Reply:
Where is the source? Can you copy the link here?

Reply:
[DiskBalancerVolume.java#L50](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/datamodel/DiskBalancerVolume.java#L50)
[DiskBalancerVolumeSet.java#L155](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/datamodel/DiskBalancerVolumeSet.java#L155)

Reply:
If `disk_densty` is not included in `query_disk_info_response`, we can add this attribute later, when we find it necessary, maybe when we have a more detailed strategy of how to use it.

Reply:
`is not included` is because acelyc111 suggest that if value can be calculated by existed value, it is no need to via `rpc`. so I delete it and re-calc on command side.